cmake/options.cmake

```diff
@@ -26,3 +26,54 @@ option(BUILD_TESTING "Whether to enable and build tests or not")
 if("${THIRD_PARTY_REPOSITORY_URL}" STREQUAL "")
   set(THIRD_PARTY_REPOSITORY_URL "https://s3.amazonaws.com/osquery-packages")
 endif()
+
+# osquery versions.
+#
+# 1. $OSQUERY_VERSION is set, use that.
+# 2. Else derive from git
+# 3. If set, append $OSQUERY_VERSION_SUFFIX
+#
+# Note that the build won't like it if the version isn't a semvar.
+
+option(OSQUERY_VERSION "Manually set the osquery version")
+option(OSQUERY_VERSION_SUFFIX "String to append to the version")
```

Comment:
I moved these because it felt cleaner to put the split and length check into options.

Reply:
I think you were referring to the list(GET)?.
The check for the length it's fine, because that's generic, but setting the CPACK variables is for packaging only.